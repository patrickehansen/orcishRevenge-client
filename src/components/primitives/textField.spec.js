import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import '@babel/polyfill';
import Provider from '../../provider';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {JssProvider} from 'react-jss';

const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`;


configure({ adapter: new Adapter() });

import TextField, {FilledTextField, OutlinedTextField, MultiTextField} from './textField';

const props = {
  label: 'test'
}

describe('Simplified TextField', () => {
  let mount;

  beforeAll(() => {
    mount = createMount();
  })

  afterAll(() => {
    mount.cleanUp();
  })

  it('Matches snapshot', () => {
    const component = mount(
      <JssProvider generateClassName={generateClassName}>
        <Provider>
          <TextField {...props}/>
        </Provider>
      </JssProvider>
    );

    expect(component).toMatchSnapshot();
  })
})

describe('Simplified FilledTextField', () => {
  let mount;

  beforeAll(() => {
    mount = createMount();
  })

  afterAll(() => {
    mount.cleanUp();
  })

  it('Matches snapshot', () => {
    const component =mount(
      <JssProvider generateClassName={generateClassName}>
        <Provider>
          <FilledTextField {...props}/>
        </Provider>
      </JssProvider>
    );

    expect(component).toMatchSnapshot();
  })
})

describe('Simplified OutlinedTextField', () => {
  let mount;

  beforeAll(() => {
    mount = createMount();
  })

  afterAll(() => {
    mount.cleanUp();
  })

  
  it('Matches snapshot', () => {
    const component =mount(
      <JssProvider generateClassName={generateClassName}>
        <Provider>
          <OutlinedTextField {...props}/>
        </Provider>
      </JssProvider>
    );

    expect(component).toMatchSnapshot();
  })
})

describe('Simplified MultiTextField', () => {
  let mount;

  beforeAll(() => {
    mount = createMount();
  })

  afterAll(() => {
    mount.cleanUp();
  })

  it('Matches snapshot', () => {
    const component =mount(
      <JssProvider generateClassName={generateClassName}>
        <Provider>
          <MultiTextField {...props}/>
        </Provider>
      </JssProvider>
    );

    expect(component).toMatchSnapshot();
  })
})