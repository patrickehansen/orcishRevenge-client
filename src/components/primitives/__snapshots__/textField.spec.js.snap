// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simplified FilledTextField Matches snapshot 1`] = `ReactWrapper {}`;

exports[`Simplified FilledTextField Matches snapshot 2`] = `ReactWrapper {}`;

exports[`Simplified FilledTextField Matches snapshot 3`] = `ReactWrapper {}`;

exports[`Simplified TextField Matches snapshot 1`] = `ReactWrapper {}`;
